crypto/dsa/dsa_backend.c

```diff
@@ -173,7 +173,10 @@ DSA *ossl_dsa_key_from_pkcs8(const PKCS8_PRIV_KEY_INFO *p8inf,
         ERR_raise(ERR_LIB_DSA, DSA_R_BN_ERROR);
         goto dsaerr;
     }
-    DSA_set0_key(dsa, dsa_pubkey, dsa_privkey);
+    if (!DSA_set0_key(dsa, dsa_pubkey, dsa_privkey)) {
+        ERR_raise(ERR_LIB_DSA, ERR_R_BN_LIB);
```

Comment:
I have no idea how we came up with these different error reasons... ERR_R_BN_LIB and also DSA_R_BN_ERROR. I would think this would not be a ERR_R_BN_LIB?

Reply:
What should it be?
"internal error" might make sense.

Reply:
That is the only place that does a raise on the setkey :) Not sure INVALID_KEY is great either..

Reply:
At some point error raising probably needs to be added all over the place.

Reply:
Agreed about more error raising.
Technically, the error here cannot be raised.  `DSA_set0_key never` fails.  At least currently, but I doubt it will change.